decompiler/pipeline/controlflowanalysis/readability_based_refinement.py

```diff
@@ -219,6 +220,28 @@ def _requirement_without_reinitialization(ast: AbstractSyntaxTree, node: Abstrac
                 return True
 
 
+def removeGuardedDoWhile(ast: AbstractSyntaxTree):
+    """ Removes a if statement which guards a do-while loop when:
+            -> there is nothing in between the if-node and the do-while-node 
+            -> the if-node has only one branch
+            -> the condition of the branch is the same as the condition of the do-while-node
+    """
+    for loop_node in list(ast.get_loop_nodes_post_order()):
+        if not isinstance(loop_node, DoWhileLoopNode) or not isinstance(loop_node.parent.parent, ConditionNode):
+            continue
+
+        condition_node = loop_node.parent.parent
+        if len(condition_node.children) != 1:
+            continue
+
+        if condition_node.true_branch:
+            condition = condition_node.true_branch.branch_condition
+        else:
+            condition = condition_node.false_branch.branch_condition
+
+        if loop_node.condition.is_equal_to(condition):
```

Comment:
One short note, I know we discussed it like this for now, but perhaps we want a follow-up issue considering the procedure as in `Helping Johnny to Analyze Malware`.